File: src/main/java/compiler/exceptions/ContainerException.java
package compiler.exceptions;

public class ContainerException extends RuntimeException{

    public ContainerException(){ super(); }
    public ContainerException(String message) { super(message); }
    public ContainerException(String message, Throwable cause) { super(message, cause); }
    public ContainerException(Throwable cause) { super(cause); }

}

File: src/main/scala/compiler/exceptions/ContainerException.scala
package compiler.exceptions

class ContainerException() extends RuntimeException {
  def this(message: String) {
    this()
    println(message)
  }

  def this(message: String, cause: Throwable) {
    this()
    println(message + " : " + cause.getStackTraceString)
  }

  def this(cause: Throwable) {
    this()
    println(cause.getStackTraceString)
  }
}